### ISSUE TITLE
Should `std` enable `rkyv` & others?

### ISSUE THREAD

**Issue by the-kenny** [paupino/rust-decimal#686]
`db-tokio-postgres` depends on `std` which, according to `README.md`, disables `no_std` support, but it seems to enable/include various optional dependencies. 

This happens because `db-tokio-postgres` depends on the `std` feature, which then activates the optional dependencies`borsh`, `rand`, `rkyv`, `serde` and `serde_json`. 

Is this intentional? I get that `db-tokio-postgres` can't work in `no_std`, but why does `std` have to enable the `rkyv` (and other) features?

**Comment by Tony-Samuels** [comment]
I don't think that is how our features set up. Do you have an example repo where you're seeing incorrect behaviour?

**Comment by Tony-Samuels** [comment]
We use `<dep>?/std` for our dependencies, which enables their `std` features *if* those dependencies are already enabled.

**Comment by paupino** [comment]
I think this is probably related to how `cargo` reports optional weak dependencies (e.g. `metadata`). This is was previously discussed a little bit in regards to weak dependencies for Rust Decimal in issue #588.

The cargo lock behavior is further discussed within the `cargo` project here: https://github.com/rust-lang/cargo/issues/10801

I think the key takeaways here:
* For what it's worth, we do seem to be using the feature correctly
* `cargo tree` should resolve the dependency tree correctly (please let us know if you have alternative results).
* The `cargo.lock` file may appear to include features not required due to the linked cargo issue.

**Comment by Sytten** [comment]
@paupino Yes that is the root cause, but most libraries now work around this limitations by adding more flags (one per dependency). You can still provide a general std flag.
Yes it is annoying to do, but polluting the lockfile of all the consumer is not great either.

Cargo is simply not going to fix this issue anytime soon and it causes a bunch of downstream problem for vendoring, security alerts, upgrades, etc.

I would strongly suggest it for rust decimal
